plan/typeinferer_test.go

```diff
@@ -122,6 +122,7 @@ func (ts *testTypeInferrerSuite) TestInferType(c *C) {
 		// Functions
 		{"version()", mysql.TypeVarString, charset.CharsetUTF8, 0},
 		{"count(c_int)", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag},
+		{"abs()", mysql.TypeNull, charset.CharsetBin, mysql.BinaryFlag},
```

Comment:
This approach will cause a mixed feature in one PR. I have to modify parser.y in order to let zero arguments passed into executor stage. It is better to file another PR to add these test you mentioned.